tests/testthat/test-formatting.R

```diff
@@ -112,3 +111,67 @@ test_that("format_na_str functionality works in get_formatted_cells (ie printing
     expect_identical(get_formatted_cells(tbl)[3, 1, drop = TRUE],
                      "Ridiculous")
 })
+
+test_that("Test matrix obtained from get_formatted_cells as is into ASCII format", {
+    # Helper function
+    deci_format <- function(x, dig) {
+        format(round(x, digits = dig), nsmall = dig)
+    }
+    
+    # Test data
+    DM2 <- DM %>% 
+        filter(ARM != levels(DM$ARM)[3]) %>% 
+        mutate(ARM = as.factor(as.character(ARM)))
+    DM2$AGE[1] <- NA # Adding one NA
+    
+    # Manually building the table
+    weird_afun <- function(x, ...) {
+        in_rows(cell_fmt = rcell(mean(x, na.rm = TRUE), format = "xx.x"),
+                no_cell_fmt = rcell(median(x, na.rm = TRUE)),
+                cell_na_str = rcell(NA, format = "xx.x"),
+                cell_na_str_no_cell_fmt = rcell(NA),
+                cell_fmt_range = rcell(range(x, na.rm = TRUE), format = "xx.x - xx.x"),
+                cell_na_str_range = rcell(c(NA, 2), format = "xx.x - xx.x", format_na_str = "bah")
+                # cell_na_str_no_cell_fmt_range = rcell(range(x))
+                )
+    }
+    
+    # Main builder
+    tbl <- basic_table() %>%
+        split_cols_by("ARM") %>%
+        analyze("AGE", weird_afun, format = "xx.xx") %>%
+        build_table(DM2)
+    
+    # Get the ASCII table
+    result <- get_formatted_cells(tbl) # Main function
+    
+    # Expected data-set is built with dplyr
+    expected <- DM2 %>% 
+        dplyr::group_by(ARM) %>%
+        summarise(
+          m_age = deci_format(mean(AGE, na.rm = TRUE), dig = 1),
```

Comment:
please use format_value (from formatters) instead of deci_format here, as the logic is similar but not identical (one of the reasons we had to write the formatting code in the first place)